server/rest/src/main/java/org/infinispan/rest/resources/ProtobufResource.java

```diff
@@ -0,0 +1,225 @@
+package org.infinispan.rest.resources;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.reactivex.rxjava3.core.Flowable;
+import org.infinispan.AdvancedCache;
+import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.commons.util.Util;
+import org.infinispan.container.entries.InternalCacheEntry;
+import org.infinispan.query.remote.ProtobufMetadataManager;
+import org.infinispan.query.remote.client.ProtobufMetadataManagerConstants;
+import org.infinispan.rest.InvocationHelper;
+import org.infinispan.rest.NettyRestResponse;
+import org.infinispan.rest.RestResponseException;
+import org.infinispan.rest.cachemanager.RestCacheManager;
+import org.infinispan.rest.framework.ContentSource;
+import org.infinispan.rest.framework.ResourceHandler;
+import org.infinispan.rest.framework.RestRequest;
+import org.infinispan.rest.framework.RestResponse;
+import org.infinispan.rest.framework.impl.Invocations;
+import org.infinispan.rest.operations.exceptions.NoDataFoundException;
+import org.infinispan.rest.operations.exceptions.NoKeyException;
+import org.infinispan.util.concurrent.CompletableFutures;
+
+import java.util.Comparator;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_JSON;
+import static org.infinispan.rest.framework.Method.DELETE;
+import static org.infinispan.rest.framework.Method.GET;
+import static org.infinispan.rest.framework.Method.POST;
+import static org.infinispan.rest.framework.Method.PUT;
+
+/**
+ * Protobuf schema manipulation Resource
+ *
+ * @author Katia Aresti
+ * @since 11
+ */
+public class ProtobufResource extends BaseCacheResource implements ResourceHandler {
+
+   private final ObjectMapper objectMapper;
+
+   public ProtobufResource(InvocationHelper invocationHelper) {
+      super(invocationHelper);
+      objectMapper = invocationHelper.getMapper();
+   }
+
+   @Override
+   public Invocations getInvocations() {
+      return new Invocations.Builder()
+            // Key related operations
+            .invocation().methods(GET).path("/v2/schema").handleWith(this::getSchemasNames)
```

Comment:
We need to rename this to ```/v2/schemas``` to conform with the other resources:
```/v2/counters/```, ```/v2/caches/```, ```/v2/tasks``` and so on. 
@oraNod also need to change all the doc references

Reply:
ok